### PR TITLE
.github: Limit CodeQL workflow to .go files

### PR DESCRIPTION
**Pull request by pchaigno**
See commits for details.

This pull request could be extended to `lint-go.yaml` but I'm not sure it's worth it currently, given that test is very fast.

### CODE REVIEW

File: .github/workflows/lint-codeql.yaml [hunk 1]
```diff
@@ -3,12 +3,20 @@ name: codeql
 on:
   push:
     branches:
+    - master
     - v1.10
     - v1.9
     - v1.8
+    paths:
+      - '*.go'
```
Comment (pchaigno) [inline .github/workflows/lint-codeql.yaml:6]: For reviewers: Are there other files that should be included here? Configuration files for CodeQL I've missed maybe?

**Comment by pchaigno** [comment]
I filled https://github.com/cilium/cilium/issues/16392 for the ConformanceKind1.19 failure.

File: .github/workflows/lint-codeql.yaml (hunk 1 above)
Comment (twpayne) [reply .github/workflows/lint-codeql.yaml:6]: Maybe add `go.mod` and `go.sum` for completeness. CodeQL itself doesn't have any config in the repo (besides the workflow file that triggers it).

**Review by twpayne — APPROVED** [review]

File: .github/workflows/lint-codeql.yaml (hunk 1 above)
Comment (pchaigno) [reply .github/workflows/lint-codeql.yaml:6]: Done.

File: .github/workflows/lint-codeql.yaml [hunk 2]
```diff
@@ -3,12 +3,24 @@ name: codeql
...
+      - 'go.mod'
+      - 'go.sum'
   pull_request:
     branches:
     - master
+    - v1.10
+    - v1.9
+    - v1.8
```
Comment (aanm) [inline .github/workflows/lint-codeql.yaml]: Shouldn't each one of these lines be in its own branch? :thinking:

Comment (pchaigno) [reply .github/workflows/lint-codeql.yaml]: I don't understand what you're asking.

Comment (aanm) [reply .github/workflows/lint-codeql.yaml]: We have a `lint-codeql.yaml` specific for each branch so the `pull_request` should be pointing only to `master`, in the v1.10 branch it should only refer to `v1.10` and so on. Does it make sense?

Comment (michi-covalent) [reply .github/workflows/lint-codeql.yaml]: it's probably easier to just do:
```
pull_request: {}
```

**Review by aanm — APPROVED** [review]

**Pull request merged**